docs/writing-stories/args.md

```diff
@@ -122,6 +122,37 @@ The `args` param is always a set of `key:value` pairs delimited with a semicolon
 
 Args specified through the URL will extend and override any default values of args specified on the story.
 
+## Mapping to complex arg values
+
+Complex values such as JSX elements cannot be serialized to the manager (e.g. the Controls addon) or synced with the URL. To work around this limitation, arg values can be "mapped" from a simple string to a complex type using the `mapping` property in `argTypes`. This works on any type of arg, but makes most sense when used with the 'select' control.
+
+```
+argTypes: {
+  label: {
+    control: {
+      type: 'select',
+      options: ['Normal', 'Bold', 'Italic']
+    },
+    mapping: {
+      Bold: <b>Bold</b>,
+      Italic: <i>Italic</i>
+    }
+  }
+}
+```
+
+Note that `mapping` does not have to be exhaustive. If the arg value is not a property of `mapping`, the value will be used directly. Keys in `mapping` always correspond to arg *values*, even when `options` is an object. Specifying `options` as an object (key-value pairs) is useful if you want to use special characters in the input label. For example:
+
+```
+{
+  control: {
+    type: 'select',
+    options: { да: 'yes', нет: 'no' }
+  },
+  mapping: { yes: 'да', no: 'нет' }
+}
```

Comment:
It would be nice if key and value were flipped on `options`, so we could use the same object for `mapping`. In fact we could make that implicit (i.e. set `mapping` to be `control.options` by default, if it's an object).

Reply:
One approach we could take it to deprecate `control.options` and introduce `options`, which uses flipped key/value pairs.

Reply:
I think the current "direction" of controls options is counter-intuitive (at least from me, I'm pretty sure in the old days of say Rails it used to be the other way round, as in many languages dict keys are pretty constrained, whereas values can be anything, so it made sense for the select's values to be the keys and the visible string to be the values). I think there's an issue about this already.

Of course we need to figure out a way to change it in a back compat way -- if we are switching it.

I think you should need to be explicit about opting in to `mapping` (it is pretty confusing otherwise that what gets passed to the component differs from what you see in all other places). My strawman would be that:

a) if an `argType` has a `mapping` and no control, we should infer a `select` control /w the `mapping` as `options`.

b) if an `argType` has a `mapping` and a `select` control, the mapping should be the default option.

WDYT?

Reply:
I'm pretty sure we can all agree key/val should be flipped, so the key becomes the arg value (before mapping). The question is how do we fix that. I see three options:

- Deprecate `control.options` and introduce `options`
- Deprecate `control.options` and introduce `control.choices` or something similar
- If `control.options` is a Map, assume key/value is flipped

As for implicitly linking `options` and `mapping`, I think we should not do any of that. With "ifs" attached, it's better to be explicit.

Reply:
Let's continue this discussion here: https://github.com/storybookjs/storybook/issues/12613